File: 21Everyday/December/yangui/372.超级次方.js
/**
 * @param {number} a
 * @param {number[]} b
 * @return {number}
 */
let base = 1337;
let dealPow = function (a, b) {
    if (b == 0) {
        return 1
    }

    a %= base
    if ((b % 2) == 0) {
        return dealPow(a * a, b / 2)
    } else {
        return a * dealPow(a, b - 1)
    }
}
var superPow = function (a, b) {

    if (b.length == 0) {
        return 1;
    }
    let last = b.pop();
    let powa = dealPow(a, last);
    let powb = dealPow(superPow(a, b), 10);

    return (powa * powb) % base;
};

File: 21Everyday/December/yangui/372超级次方/372.超级次方.py
base = 1337
class Solution:

    def superPow(self, a: int, b: List[int]) -> int:
        if len(b) == 0:
            return 1 

        last = b.pop()
        powa = self.dealPow(a, last)
        powb = self.dealPow(self.superPow(a,b), 10)

        return (powa * powb) % base

    def dealPow(self, a: int, b: int):
        if b == 0:
            return 1
        a %= base
        if (b % 2) == 0:
            return self.dealPow(a * a, b / 2)
        else:
            return a * self.dealPow(a, b-1)
